### PR TITLE
OcdFileImport: Handle emtpy area holes

### PR DESCRIPTION
**Pull request by dl3sdo**
There are .ocd files that erroneously contain an empty area hole, i.e., there are two subsequent points with the Ocd::OcdPoint32::FlagHole property.
If such a malformed .ocd file is imported Mapper will crash if the area object containing the empty hole is selected. 
To fix this, the first of these .ocd points needs to be ignored. Since setPointFlags() applies the HolePoint property to the last point instead of the first point of the next part, the consequence is that instead of removing the first point of these two .ocd points, the second point of the two Mapper points with the HolePoint property needs to be removed.

### CODE REVIEW

**Review by dl3sdo — COMMENTED** [review]
Removing empty area holes should not be restricted to just one but be extended to multiple subsequent ones by replacing the `if` statement by a `while` loop.

File: src/fileformats/ocd_file_import.cpp [hunk 1]
```diff
@@ -2153,6 +2153,17 @@ void OcdFileImport::fillPathCoords(OcdImportedPathObject *object, bool is_area,
 		size_t start = 0;
 		for (size_t i = 0; i < object->coords.size(); ++i)
 		{
+			// There are .ocd files that erroneously contain an empty area hole, i.e., there are two subsequent points 
+			// with the Ocd::OcdPoint32::FlagHole property. To fix this, the first of these points needs to be ignored.
+			// Since setPointFlags() applies the HolePoint property to the last point instead of the first point of the next part,
+			// the consequence is that instead of removing the first point of these two .ocd points,
+			// the second point of the two Mapper points with the HolePoint property needs to be removed.
+			// Using a loop ensures that multiple subsequent empty area holes are removed.
+			while (is_area && i < object->coords.size() - 1 && object->coords[i].isHolePoint() && object->coords[i+1].isHolePoint())
```
Comment (dg0yt) [inline src/fileformats/ocd_file_import.cpp:2162]: I think it might be better to use a separate loop (or STL algorithm) then to nest loops.

**Review by dg0yt — COMMENTED** [review]
Maybe I need some real or synthetical test data here.

**Comment by dl3sdo** [comment]
Kai, I'll send you files via email.

**Comment by dl3sdo** [comment]
Superseded by #2224 and #2225

**Pull request closed without merging**